local.tf

```diff
@@ -19,8 +19,8 @@ locals {
   worker_group_count                 = length(var.worker_groups)
   worker_group_launch_template_count = length(var.worker_groups_launch_template)
 
-  default_ami_id_linux   = data.aws_ami.eks_worker.id
-  default_ami_id_windows = data.aws_ami.eks_worker_windows.id
+  default_ami_id_linux   = lookup(var.workers_group_defaults, "ami_id", data.aws_ami.eks_worker.id)
+  default_ami_id_windows = lookup(var.workers_group_defaults, "ami_id", data.aws_ami.eks_worker_windows.id)
```

Comment:
@barryib not sure I follow with using coalesce instead of lookup. With lookup you specify a key to get a value or default to something else, coalece picks first non null element... not sure how this is going to solve the problem of an empty query result from the ami data source... it might be a downstream problem pertaining to our account...

Reply:
@barryib I've just implemented your improvement request. Also found out why it tried to use null instead of the default from the data, wasn't checking for empty provided value.
Works as advertised and tested a few permutations. Let me know if you need more changes. Thanks for your input!